payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultPaymentAuthenticatorRegistry.kt

```diff
@@ -78,15 +98,27 @@ internal class DefaultPaymentAuthenticatorRegistry @Inject internal constructor(
         activityResultCaller: ActivityResultCaller,
         activityResultCallback: ActivityResultCallback<PaymentFlowResult.Unvalidated>
     ) {
-        paymentAuthenticatorMap.values.forEach {
+        allAuthenticators.forEach {
             it.onNewActivityResultCaller(activityResultCaller, activityResultCallback)
         }
+        paymentRelayLauncher = activityResultCaller.registerForActivityResult(
+            PaymentRelayContract(),
+            activityResultCallback
+        )
+        paymentBrowserAuthLauncher = activityResultCaller.registerForActivityResult(
+            PaymentBrowserAuthContract(),
+            activityResultCallback
+        )
     }
 
     override fun onLauncherInvalidated() {
-        paymentAuthenticatorMap.values.forEach {
+        allAuthenticators.forEach {
             it.onLauncherInvalidated()
         }
+        paymentRelayLauncher?.unregister()
+        paymentBrowserAuthLauncher?.unregister()
+        paymentRelayLauncher = null
+        paymentBrowserAuthLauncher = null
```

Comment:
Instead of relying on this, can this class be made lifecycle-aware?

Reply:
hmm that's an interesting idea, I just did some draft locally - 
* currently we're manually calling `onNewActivityResultCaller` and `onLauncherInvalidated` on certain lifecycle events. (DefaultFlowController's [init](https://github.com/stripe/stripe-android/blob/cdd2def71f8732c5ccf164a51ccf2501335dab06/payments-core/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt#L125) and PaymentSheetActivity's [onCreate](https://github.com/stripe/stripe-android/blob/a9bdc145ee5ec64f94a7406fff396f0eb847a944/payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt#L98) and [onDestroy](https://github.com/stripe/stripe-android/blob/a9bdc145ee5ec64f94a7406fff396f0eb847a944/payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetActivity.kt#L164)).
* I guess what you meant here is to trigger the clean up during `ON_DESTROY` lifecycle event, to achieve that, there are two ways
  1. pass a `lifeCycle` object in `onNewActivityResultCaller` as a new parameter, we still write the launchers as is, but create a observer to clean them up:
    ```
    fun onNewActivityResultCaller(activityResultCaller, activityResultCallback, lifecycle) {
        // ... still set up the launchers
        lifecycle.addObserver {
            @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
            fun onDestroy() {
                // clean up 
            }
        }
      }
    ```
  2. pass three more params, `lifeCycle`, `activityResultCallerProvider` and `activityResultCallbackProvider` as a new constructor parameter of `DefaultPaymentAuthenticatorRegistry`. During init, create an observer that sets the launchers in ON_CREATE and clear them in ON_DESTROY. Now `PaymentAuthenticatorRegistry` no longer need to implement [ActivityResultLancherHost](https://github.com/stripe/stripe-android/blob/142a12534f1c80f0691647566f06dbf7a15f0ea2/payments-core/src/main/java/com/stripe/android/payments/core/authentication/PaymentAuthenticatorRegistry.kt#L8)

Solution i isn't great because it makes half(clean up) of it lifecycle ware and another half(write) not, I think if we were to implement this, solution ii is the right way to go, we could probably even create some class to hold the 3 params. This change seems pretty big, I'm happy to do it in a follow up, wdty?

Reply:
on a 2nd thought just realized solution ii is not feasible either because `DefaultPaymentAuthenticatorRegistry` actually lives longer than an `Activity`/`LifeCycleOwner`
e.g `PaymentSheetViewModel` holds a `PaymentController`, which holds a `DefaultPaymentAuthenticatorRegistry`, the same viewModel instance will be reused when `PaymentSheetActivity` is recreated

Reply:
What about making the lifecycle of `DefaultPaymentAuthenticatorRegistry` shorter by creating a new one every time `onNewActivityResultCaller` would be called? So the `ActivityResultCaller` and callback never change?
I'm just trying to think of possible simplifications of this code, having to manage this manually seems complex and error-prone.

Reply:
this pair of methods exist mainly for compatibility reasons, because `StripePaymentController` hosts a registry and has longer lifecycles than `Activity`, so we'll need to also make a new `StripePaymentController` instance if we were to make that change.

In the new `PaymentLauncher`, since we control the host activity, this pair will be guaranteed to called exactly once. but again we can't change it yet because it's still being used by `StripePaymentController` :(